src/main/java/hudson/remoting/Engine.java

```diff
@@ -839,12 +840,15 @@ private boolean succeedsWithRetries(java.util.concurrent.Callable<Boolean> condi
     @SuppressFBWarnings(
             value = {"URLCONNECTION_SSRF_FD"},
             justification = "url is provided by the user, and we are trying to connect to it")
-    private Boolean pingSuccessful() throws MalformedURLException {
+    private Boolean pingSuccessful(SSLContext sslContext) throws MalformedURLException {
         // Unlike JnlpAgentEndpointResolver, we do not use $jenkins/tcpSlaveAgentListener/, as that will be
         // a 404 if the TCP port is disabled.
         URL ping = new URL(hudsonUrl, "login");
         try {
             HttpURLConnection conn = (HttpURLConnection) ping.openConnection();
+            if (conn instanceof HttpsURLConnection && sslContext != null) {
+                ((HttpsURLConnection) conn).setSSLSocketFactory(sslContext.getSocketFactory());
```

Comment:
BTW https://github.com/jenkinsci/remoting/blob/dbefbe0b56798369075e1ee0cce5a06c1cf7d116/pom.xml#L70-L71 (#749) surely time by now? 2.479.1+ 2024-10-30 so I am not sure what purpose is served by continuing to publish new `remoting` releases that can connect to older controllers (missing multiple rounds of security fixes).

> wait until at least the end of October

Reply:
Java 17 is the new minimum so you should be able to do:
```suggestion
            if (conn instanceof HttpsURLConnection httpsCon && sslContext != null) {
                httpsCon.setSSLSocketFactory(sslContext.getSocketFactory());
```